libGitWrap/Tree.cpp

```diff
@@ -81,19 +81,38 @@ namespace Git
         return PrivatePtr(new Tree::Private(d->repo(), subObject));
     }
 
-    DiffList Tree::diffToTree(Result& result , Tree newTree)
+    /**
+     * @brief           Diff this tree to a newer tree.
+     *
+     * @param[in,out]   result      A Result object; see @ref GitWrapErrorHandling
+     *
+     * @param[in]       newTree     is considered as the "new tree" to compare with
+     *
+     * @return          the differences as @ref DiffList object
+     *
+     * The calling Tree object is considered as the "old tree" and @p newTree
+     * is the "new tree" to compare with. When exchanging both trees, the
+     * result will be inversed.
+     *
+     * An invalid Tree object is is considered an empty tree.
+     */
+    DiffList Tree::diffToTree(Result& result, Tree newTree)
     {
-        GW_D_CHECKED(Tree, DiffList(), result)
+        // d is allowed to be invalid
```

Comment:
Actually, this is why I want to merge this PR now and open a new one :smile:. Following the idiom: "make it work; make it right; make it fast".

Actually I was thinking the same directly, when writing it. And ... when you look i.e. into `Tree::diffToIndex`, this won't work at all (repo and index are taken from the Tree object). While libgit2 OTOH accepts a NULL tree as an empty tree here, too.

I think actually of opening another PR after this one is merged, adding a `Git::Diff` class, containing probably only static calls like: `Git::Diff::treeToTree( result, oldTree, newTree )`. And then I want to deprecate "everything else" (This makes more sense than `Repository::diffTreeToTree´). And it also makes our public API more convenient to libgit2. But ... this is another bigger topic. And I want to be able to present something to the public as soon as possible to get more acceptance (hopefully devs also).

Conclusion:
What I really like to do is to make GitWrap more to what its name is saying - a light Qt-Wrapper for libgit2. Nothing more, nothing less. Unlike in the past, lg2 is very feature complete and its API is more robust now. Thus, writing the wrappers has become easier -> means shorter development cycles. The pretty much only exception is the `Git::Operation framework`, that makes GW more powerful and flexible.